presto-client/src/main/java/com/facebook/presto/client/FixJsonDataUtils.java

```diff
@@ -201,4 +227,51 @@ private static Object fixValue(TypeSignature signature, Object value)
                 return value;
         }
     }
+
+    private static Map<String, String> fixObjectFromMap(String value)
+    {
+        List<String> entries = fixObjectFromList(value);
+        Map<String, String> fixedValue = new HashMap<>();
+        for (String entry : entries) {
+            int bracketCount = 0;
+            for (int i = 0; i < entry.length(); i++) {
+                if (entry.charAt(i) == '{' || entry.charAt(i) == '[') {
+                    bracketCount += 1;
+                }
+                else if (entry.charAt(i) == '}' || entry.charAt(i) == ']') {
+                    bracketCount -= 1;
+                }
+                else if (entry.charAt(i) == '=' && bracketCount == 0) {
+                    String keyString = entry.substring(0, i);
+                    String valString = entry.substring(i + 1);
+                    fixedValue.put(keyString, valString);
+                    break;
+                }
+            }
+        }
+        return fixedValue;
+    }
+    private static List<String> fixObjectFromList(String value)
+    {
+        int bracketCount = 0;
+        List<String> fixedValue = new ArrayList<>();
+        int j = 1;
+        for (int i = 0; i < value.length(); i++) {
+            if (value.charAt(i) == '{' || value.charAt(i) == '[') {
+                bracketCount += 1;
+            }
+            else if ((value.charAt(i) == ']' || value.charAt(i) == '}') && bracketCount == 1) {
```

Comment:
What if the key or value contains an escaped bracket?  Can you add a test for this?